text/0000-inline-asm.md

```diff
@@ -816,6 +818,14 @@ Inline assembly is a difficult feature to implement in a compiler backend. While
 
 Because `{}` are used to denote operand placeholders in the template string, actual uses of braces in the assembly code need to be escaped with `{{` and `}}`. This is needed for AVX-512 mask registers and ARM register lists.
 
+## Post-monomorphization errors
+
+Since the code generated by `asm!` is only evaluated late in the compiler back-end, errors in the assembly code (e.g. invalid syntax, unrecognized instruction, etc) are reported during code generation unlike every other error generated by rustc. In particular this means that:
+- Since `cargo check` skips code generation, assembly code is not checked for errors.
+- `asm!` blocks that are determined to be unreachable are not checked for errors. This can even vary depending on the optimization level since inlining provides more opportunities for constant propagation.
+
+However there is a precedent in Rust for post-monomorphization errors: linker errors. Code which references a non-existent `extern` symbol will only cause an error at link-time, and this can also vary with optimization levels as dead code elimination may removed the reference to the symbol before it reaches the linker.
```

Comment:
Also `mem::transmute` with different type sizes.

Reply:
Actually `mem::transmute` checks are done pre-monomorphization: transmuting generic types is not allowed.

Reply:
You are right. Layout errors like size too big are done post-monomorphization though:

```rust
pub fn a<T>(v: [u16; usize::max_value()]) {
    let x = v;
}
```

is fine, while

```rust
pub fn a(v: [u16; usize::max_value()]) {
    let x = v;
}
```

isn't.